tests/nightly/test_large_vector.py

```diff
@@ -27,12 +28,508 @@
+
+#     sym = mx.sym.SoftmaxOutput(data=x, label=label, ignore_label=0,
+#                                use_ignore=False)
+#     ex = sym.bind(ctx=default_context(), args={'x': x_nd, 'label': label_nd},
+#                   args_grad={'x': grad_x})
+
+#     ex.forward(is_train=True)
+#     softmax_out = ex.outputs[0][0].asnumpy()
+#     expected_softmax_out = (1/SMALL_Y)*mx.nd.ones((SMALL_Y)).asnumpy()
+#     assert np.isclose(softmax_out, expected_softmax_out).all()
+
+#     ex.backward(is_train=True)
+#     grad_out = ex.grad_arrays[0][0].asnumpy()
+#     k = int(label_nd[0].asscalar())
+#     expected_grad_out = np.zeros((SMALL_Y,))
+#     expected_grad_out[k] = -1
+#     assert np.isclose(grad_out - softmax_out, expected_grad_out).all()
+
+
+# TODO: correctness of prelu (currently flaky)
+def test_leaky_relu():
+    a = -1*mx.nd.ones((LARGE_X, SMALL_Y))
+
+    def test_leaky():
+        res = mx.nd.LeakyReLU(a, act_type="leaky", slope=0.3)
+        assert res[-1][-1].asnumpy() == 0.3*a[-1][-1].asnumpy()
+
+    def test_elu():
+        res = mx.nd.LeakyReLU(a, act_type="elu", slope=0.3)
+        assert res[-1][-1].asnumpy() == 0.3*(np.exp(a[-1][-1].asnumpy())-1)
+
+    def test_selu():
+        lam = 1.0507009873554804934193349852946
+        alpha = 1.6732632423543772848170429916717
+        res = mx.nd.LeakyReLU(a, act_type="selu")
+        assert res[-1][-1].asnumpy() == (lam * alpha * (np.exp(a[-1][-1].asnumpy())-1))
+
+    def test_rrelu():
+        lower = 0.125
+        upper = 0.333999991
+        res = mx.nd.LeakyReLU(a, act_type="rrelu")
+        assert res[-1][-1].asnumpy() == (lower + upper) / 2 * a[-1][-1].asnumpy()
+
+    test_leaky()
+    test_elu()
+    test_selu()
+    test_rrelu()
+
+
+def test_layer_norm():
+    dtype = np.float32
+    forward_check_eps = 1E-3
+    axis = 0
+    eps = 1E-5
+    in_shape = (LARGE_X,)
+
+    def npy_layer_norm(data, gamma, beta, axis=0, eps=1E-5):
+        broadcast_shape = [1 for _ in range(data.ndim)]
+        broadcast_shape[axis] = data.shape[axis]
+        mean = data.mean(axis=axis, keepdims=True).astype(dtype)
+        var = data.var(axis=axis, keepdims=True).astype(dtype)
+        std = np.sqrt(var + dtype(eps)).astype(dtype)
+        out = np.reshape(gamma, broadcast_shape) * (data - mean) / std + \
+              np.reshape(beta, broadcast_shape)
+        return out
+    data = np.random.normal(0, 1, in_shape).astype(dtype)
+    gamma = np.random.normal(0, 1, (in_shape[axis],)).astype(dtype)
+    beta = np.random.normal(0, 1, (in_shape[axis],)).astype(dtype)
+    mx_out = nd.LayerNorm(data, gamma, beta, axis, eps)
+    np_out = npy_layer_norm(data, gamma, beta, axis, eps)
+    assert_almost_equal(np_out, mx_out.asnumpy(), forward_check_eps,
+                        forward_check_eps)
+
+
+# TODO: correctness of dropout
+# currently only test for dropout to work
+# since testing for correctness involves flakiness issue #14288
+def test_dropout():
+    shape = (LARGE_X, )
```

Comment:
use 1D shape.

Reply:
In case of dropout, because it is in symbolic
when I try to bind with 1D number
```
RuntimeError: simple_bind error. Arguments:
data: 10
[17:09:03] ../src/executor/../common/exec_utils.h:391: InferShape pass cannot decide shapes for the following arguments (-1 means unknown dimensions). Please consider providing them as inputs:
data: None,
```

But this passes
```
>>> exe = y.simple_bind(ctx=mx.cpu(), data=(10,))
[17:09:41] ../src/executor/graph_executor.cc:1942: Subgraph backend MKLDNN is activated.
```

I'm guessing it expects parenthesis to know it's a shape in case of simple_bind
http://mxnet.incubator.apache.org/versions/master/api/python/symbol/symbol.html#mxnet.symbol.Symbol.simple_bind

Reply:
Also found that if it's (10,) or (10) ultimately NDArray is still 1D
```
>>> mx.nd.uniform(shape=10)

[0.5488135  0.5928446  0.71518934 0.84426576 0.60276335 0.8579456
 0.5448832  0.8472517  0.4236548  0.6235637 ]
<NDArray 10 @cpu(0)>
>>> mx.nd.uniform(shape=(10,))

[0.6458941  0.3843817  0.4375872  0.2975346  0.891773   0.05671298
 0.96366274 0.2726563  0.3834415  0.47766513]
<NDArray 10 @cpu(0)>
```